### PR TITLE
Update `ButtonGroup` to use `Beta::Button`

### PR DESCRIPTION
**Pull request by langermank**
### Description

- Removes deprecated Button component
- Adds CSS for `ButtonGroup` class
- Add preview for split button ActionMenu scenario (WIP)

### Integration

> Does this change require any updates to code in production?

Maybe, this might be kind of a breaking change? This is used 18 times in production and I'm happy to roll it out with any fixes. 

Replace `variant: :small` with `size: :small` https://github.com/search?q=repo%3Agithub%2Fgithub+Primer%3A%3ABeta%3A%3AButtonGroup&type=code

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews

### CODE REVIEW

**Comment by jonrohan** [comment]
> Maybe, this might be kind of a breaking change? This is used 18 times in production and I'm happy to roll it out with any fixes.

Taking a quick look the only thing it looks like we'll need to fix is `variant: ` is used but not in too many places, I think this could be doable https://github.com/search?q=repo%3Agithub%2Fgithub+Primer%3A%3ABeta%3A%3AButtonGroup&type=code

**Comment by langermank** [comment]
> Taking a quick look the only thing it looks like we'll need to fix is `variant: ` is used but not in too many places, I think this could be doable https://github.com/search?q=repo%3Agithub%2Fgithub+Primer%3A%3ABeta%3A%3AButtonGroup&type=code

Yeah, and it was pointing to `size` for some reason? I guess I could just leave it in and let it be.

**Review by jonrohan — CHANGES_REQUESTED** [review]
Since https://github.com/primer/view_components/pull/1991 is already getting really full, and I know we'll need to make updates for this change, let's hold off merging this until after `0.1.9` merges.

File: app/components/primer/beta/button_group.rb [hunk 1]
```diff
@@ -8,45 +8,45 @@ class ButtonGroup < Primer::Component
 
       # Required list of buttons to be rendered.
       #
-      # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::ButtonComponent) %> except for `size` and `group_item`.
+      # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::Beta::Button) %>
       renders_many :buttons, lambda { |**kwargs|
-        kwargs[:group_item] = true
-        kwargs[:size] = @size
+                               kwargs[:size] = @size
```
Comment (camertron) [inline app/components/primer/beta/button_group.rb:13]: What's going on with the indentation of this lambda?

File: previews/primer/beta/button_group_preview/action_menus.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<%= render(Primer::Beta::ButtonGroup.new()) do |component| %>
```
Comment (camertron) [inline previews/primer/beta/button_group_preview/action_menus.html.erb:1]:
```suggestion
<%= render(Primer::Beta::ButtonGroup.new) do |component| %>
```

**Review by camertron — CHANGES_REQUESTED** [review]

File: app/components/primer/beta/button_group.rb (hunk 1 above)
Comment (langermank) [reply app/components/primer/beta/button_group.rb]: fixed

File: test/components/beta/button_group_test.rb [hunk 3]
```diff
@@ -24,16 +24,14 @@ def test_renders_button_with_props
       component.with_button { "Button" }
       component.with_button(scheme: :primary) { "Primary" }
       component.with_button(scheme: :danger) { "Danger" }
```
Comment (jonrohan) [inline test/components/beta/button_group_test.rb:26]: You'll need to pass the scheme to ButtonGroup now

File: app/components/primer/beta/button_group.rb [hunk 4]
```diff
@@ -8,45 +8,45 @@ class ButtonGroup < Primer::Component
...
-      # @param variant [Symbol] DEPRECATED. <%= one_of(Primer::ButtonComponent::SIZE_OPTIONS) %>
-      # @param size [Symbol] <%= one_of(Primer::ButtonComponent::SIZE_OPTIONS) %>
+      # @param scheme [Symbol] DEPRECATED. <%= one_of(Primer::Beta::Button::SCHEME_OPTIONS) %>
+      # @param size [Symbol] <%= one_of(Primer::Beta::Button::SIZE_OPTIONS) %>
       # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-      def initialize(variant: nil, size: Primer::ButtonComponent::DEFAULT_SIZE, **system_arguments)
-        @size = variant || size
+      def initialize(scheme: Primer::Beta::Button::DEFAULT_SCHEME, size: Primer::Beta::Button::DEFAULT_SIZE, **system_arguments)
+        @size = size
+        @scheme = scheme
```
Comment (jonrohan) [inline app/components/primer/beta/button_group.rb:44]: You might want to use fetch or fallback here and the Button scheme and size options

**Review by camertron — APPROVED** [review]

**Comment by langermank** [comment]
This is ready to merge, but I'd like to be available to help roll it out and make dotcom changes. I will be out May 22-26.

**Review by jonrohan — APPROVED** [review]
The older release is out, 🚀

**Pull request merged**